src/lib/components/simple/lists/SimpleTable.svelte

```diff
@@ -1,13 +1,35 @@
 <script context="module" lang="ts">
+    
+  export type HeaderColumnDateFormat = "toISO" | "toISODate" | "toLocaleString" | 'setLocale' | 'fromISO'
+  
+  export type ColumnDate = {
+      key: "date",
+      params: {
+        format?: HeaderColumnDateFormat,
+      }
+  }
+
+  export type ColumnIcon = {
+      key: "icon",
+      params: {
+          name: string,
+          color: string,
+          size: string
+      }
+  }
+
+  export type HeaderTypeAdvanced =  ColumnDate | ColumnIcon
+  export type HeaderType = "boolean" | "string" | "number"  | "custom" | 'checkbox';
+
   export type Header = {
-    value: string
-    label: string
-    type: "boolean" | "string" | "number" | "date" | "custom"
-    width?: string
-    minWidth?: string
-    sortable?: boolean
+    value: string;
+    label: string;
+    type: HeaderType | HeaderTypeAdvanced; 
```

Comment:
Qua ci sta che usi il type come oggetto ma farei a questo punto che tutti i tipi sono degli oggetti per queste ragioni:

1. Rischiamo di aggiungere della logica in più per controllare se il campo è di tipo stringa o oggetto
2. Potrebbero incasinarsi i tipi nel caso avessimo bisogno di fare dei Literals e il compiler potrebbe non essere abbastanza intelligente da distinguere i Literals in base agli if
3. La situazione diverrebbe esteticamente più carina